src/vm/tests/mock_tests/mock_test_allocate_with_re_enable_collection.rs

```diff
@@ -31,29 +37,23 @@ pub fn allocate_with_re_enable_collection() {
             );
             assert!(!addr.is_zero());
 
-            // Disable GC. So we can keep allocate without triggering a GC.
-            memory_manager::disable_collection(fixture.mmtk());
+            // In the next allocation GC is disabled. So we can keep allocate without triggering a GC.
             // Fill up the heap
-            let _ = memory_manager::alloc(
-                &mut fixture.mutator,
-                MB >> 1,
-                8,
-                0,
-                AllocationSemantics::Default,
-            );
+            let _ =
+                memory_manager::alloc(&mut fixture.mutator, MB, 8, 0, AllocationSemantics::Default);
 
-            // Enable GC again.
-            memory_manager::enable_collection(fixture.mmtk());
-            // Attempt another allocation. This will trigger GC.
+            // Attempt another allocation. This will trigger GC since GC is enabled again.
             let addr =
                 memory_manager::alloc(&mut fixture.mutator, MB, 8, 0, AllocationSemantics::Default);
             assert!(!addr.is_zero());
         },
         || {
-            // This is actually redundant, as we defined block_for_gc for this test.
-            // This just demostrates that we can check if the method is called.
+            // This ensures that block_for_gc is called for this test, and that the second allocation
+            // does not trigger GC since we expect is_collection_enabled to be called three times.
             read_mockvm(|mock| {
-                assert!(mock.block_for_gc.is_called());
+                assert!(
+                    mock.block_for_gc.is_called() && mock.is_collection_enabled.call_count() == 3
+                );
```

Comment:
I suggest splitting it into two assertions so that when one of them fails, we know exactly which one failed.

```suggestion
                assert!(mock.block_for_gc.is_called());
                assert!(mock.is_collection_enabled.call_count() == 3);
```